### PR TITLE
Qm adhras

### PR DESCRIPTION
**Pull request by Bijaya-Maitri**
## Describe your changes
Please include a summary of any changes.

## How has this been tested?
Please describe the tests you ran to verify your changes.  Provide instructions or code to reproduce output.

## Reviewer focus
Please summarize the specific items you’d like the reviewer(s) to look into.

## Checklist before requesting a review
- [ ] I have added at least one Github label to this PR (bug, enhancement, breaking change,...)
- [ ] My code follows [style guidelines](https://thetuvaproject.com/guides/contributing/style-guide)
- [ ] (New models) [YAML files](https://github.com/tuva-health/tuva/blob/main/models/hcc_suspecting/hcc_suspecting_models.yml) are categorized by sub folder and models listed in alphabetical order
- [ ] (New models) I have added a [config](https://github.com/tuva-health/tuva/blob/main/models/hcc_suspecting/final/hcc_suspecting__list.sql) to each new model to enable it for claims and/or clinical data
- [ ] (New models) I have added the variable `tuva_last_run` to the final output
- [ ] (Optional) I have recorded a Loom to explain this PR

### Package release checklist
- [ ] I have updated [dbt docs](https://www.notion.so/tuvahealth/Building-dbt-Docs-16df2f00df244f29b9d6756d8adbc2d9)
- [ ] I have updated the version number in the `dbt_project.yml`

### CODE REVIEW

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras__performance_period.sql [hunk 1]
```diff
@@ -0,0 +1,85 @@
+{{ config(
+     enabled = var('quality_measures_enabled',var('claims_enabled',var('clinical_enabled',var('tuva_marts_enabled',False))))
+     | as_bool
+   )
+}}
+
+{%- set measure_id -%}
+
+(select id
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras__performance_period.sql]: C1: Make a newline after (

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras__performance_period.sql [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+from {{ ref('quality_measures__measures') }}
+where id = 'ADH-RAS')
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras__performance_period.sql]: C2: Add a newline before )

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras__performance_period.sql [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+{%- endset -%}
+
+{%- set measure_name -%}
+
+(select name
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras__performance_period.sql]: C1/c2

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras__performance_period.sql [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+
+/* , period_begin as (
+
+    select
+          performance_period_end
+        ,  {{ date_trunc('year', "performance_period_end") }}  as performance_period_begin
+    from period_end
+)
+*/
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras__performance_period.sql]: Remove comments

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    select
+          code
+        , code_system
+    from {{ ref('quality_measures__value_sets') }}
+    where lower(concept_name) = 'pqa rasa medications'
+
+)
+
+, pharmacy_claim  as (
+    select
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql:31]: add newline

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+
+    select
+          patient_id
+        , dispensing_date
+        , days_supply
+        , performance_period_begin
+        , performance_period_end
+    from patient_with_claim as pc
+    inner join performance_period as pp
+    on pc.dispensing_date between pp.performance_period_begin and pp.performance_period_end
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql]: follow one-tab indent for joining conditions on every new line

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+
+, patient_with_claim as (
+
+    select
+          pc.patient_id
+        , dispensing_date
+        , days_supply
+        , ndc_code
+    from pharmacy_claim as pc
+    inner join visit_codes as vc
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql]: dont alias for short names like this, 
also use readable words for aliasing

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+    on pc.ndc_code = vc.code
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql]: also match with code_system = 'ndc'

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+        , dispensing_date
+        , days_supply
+        , row_number() over (partition by patient_id order by dispensing_date) as row_number
+    from patient_within_performance_period
+
+)
+
+, patient_with_first_dispensing_date as (
+
+    select distinct
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql]: Confirm if distinct is really needed here

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+, patient_with_first_fill_greater_than_90 as (
+
+    select
+          pp.patient_id
+        , pp.dispensing_date
+        , fdd.first_dispensing_date
+        , pp.days_supply
+    from patient_within_performance_period as pp
+    inner join patient_with_first_dispensing_date as fdd
+    on pp.patient_id = fdd.patient_id
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql]: indentation

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+
+, patient_with_first_fill_greater_than_90 as (
+
+    select
+          pp.patient_id
+        , pp.dispensing_date
+        , fdd.first_dispensing_date
+        , pp.days_supply
+    from patient_within_performance_period as pp
+    inner join patient_with_first_dispensing_date as fdd
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql]: dont alias this

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql [hunk 12]
```diff
@@ -0,0 +1,194 @@
...
+    on pp.patient_id = fdd.patient_id
+    where {{ datediff('first_dispensing_date', 'performance_period_end', 'day') }} > 89
+
+)
+
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql:113]: Remove extra newline

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql [hunk 13]
```diff
@@ -0,0 +1,194 @@
...
+
+, patient_with_more_than_1_prescriptions as (
+
+    select distinct
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql]: distinct necessity confirm

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql [hunk 14]
```diff
@@ -0,0 +1,194 @@
...
+      patient_id
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql]: one tab indent

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql (hunk 11 above)
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql]: This join might not be needed as one table filtered of the other

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql [hunk 15]
```diff
@@ -0,0 +1,194 @@
...
+      from patient_with_row_number
+      where row_number = 2
+
+)
+
+, patient_with_mt_1_prescriptions_and_first_fill_gt_90 as (
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql]: please write english_words for cte names (readability)

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql [hunk 16]
```diff
@@ -0,0 +1,194 @@
...
+
+    select
+          patient_id
+        , sex
+        , birth_date
+        , death_date
+    from {{ ref('quality_measures__stg_core__patient') }}
+
+)
+
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql]: This cte is not referenced anywhere in the model

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql [hunk 17]
```diff
@@ -0,0 +1,194 @@
...
+, patient_with_age as (
+
+    select
+          pp.patient_id
+    from {{ref('quality_measures__stg_core__patient')}} as p
+    inner join patient_within_performance_period as pp
+    on p.patient_id = pp.patient_id
+    where (floor({{ datediff('birth_date', 'pp.performance_period_begin', 'hour') }} / 8760.0)) > 17
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql]: make an extra cte for computations like this

Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql]: ensure dead patients are not included

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql [hunk 18]
```diff
@@ -0,0 +1,194 @@
...
+    , first_dispensing_date
+    , days_supply
+    , performance_period_begin
+    , performance_period_end
+    , measure_id
+    , measure_name
+    , measure_version
+    , denominator_flag
+    , '{{ var('tuva_last_run')}}' as tuva_last_run
+from add_data_types
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql]: add newline on EOF

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql [hunk 19]
```diff
@@ -0,0 +1,194 @@
...
+    on p.patient_id = pp.patient_id
+    where (floor({{ datediff('birth_date', 'pp.performance_period_begin', 'hour') }} / 8760.0)) > 17
+
+)
+
+, qualifying_patients as (
+
+    select
+          pp90.patient_id
+        , pp90.dispensing_date
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql]: are all dispensing dates needed on down lineage models?

File: models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql [hunk 20]
```diff
@@ -0,0 +1,194 @@
...
+        , pp90.first_dispensing_date
+        , pp90.days_supply
```
Comment (utsavpaudel) [inline models/quality_measures/intermediate/adh-ras_medication_adherence_for_hypertension/quality_measures__int_adhras_denominator.sql]: same for days_supply

**Review by utsavpaudel — COMMENTED** [review]
Hey @BijayaShres, good job on first model completion. Just a few requests for semantic changes and reducing computation workload.

**Comment by Bijaya-Maitri** [comment]
Thank You so much @utsavpaudel  for the review

**Pull request closed without merging**